include/tests/TestResult.h

```diff
@@ -17,12 +17,21 @@ struct TestResult {
       : name(in.stem()), pass(pass), error(error), diff(diff) {}
 
   // Info about result.
-  const fs::path name;
-  const bool pass;
-  const bool error;
-  const std::string diff;
-};
+  fs::path name;
+  bool pass;
+  bool error;
+  std::string diff;
+
+  TestResult clone() { return TestResult(this->name, this->pass, this->error, this->diff); }
 
+  void swap(TestResult &other) {
```

Comment:
Is there any reason that this can't just be `std::swap` on two instances of `TestResult` to begin with?